server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -3711,6 +3711,11 @@ public SSHKeyPair createSSHKeyPair(final CreateSSHKeyPairCmd cmd) {
         final SSHKeysHelper keys = new SSHKeysHelper(sshKeyLength.value());
 
         final String name = cmd.getName();
+        
+        if (StringUtils.isBlank(name)) {
+            throw new InvalidParameterValueException("Please specify a valid name for the key pair. The key name can't be empty");
+        }
+        
```

Comment:
can you move this validation above the validation that queries the DB above?

Reply:
Thanks for your comment. Moved it.